In [666]:
import pandas as pd
import bokeh
import pandas_bokeh


In [667]:
# Embedding plots in Jupyter Notebook
pandas_bokeh.output_notebook()

Loading BokehJS ...

# Data overview

In [677]:
df = pd.read_csv('hospital-charges.csv')
df.head(3)

DRG Definition  Provider Id  \
0  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10001   
1  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10005   
2  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10006   

                      Provider Name     Provider Street Address Provider City  \
0  SOUTHEAST ALABAMA MEDICAL CENTER      1108 ROSS CLARK CIRCLE        DOTHAN   
1     MARSHALL MEDICAL CENTER SOUTH  2505 U S HIGHWAY 431 NORTH          BOAZ   
2    ELIZA COFFEE MEMORIAL HOSPITAL          205 MARENGO STREET      FLORENCE   

  Provider State  Provider Zip Code Hospital Referral Region Description  \
0             AL              36301                          AL - Dothan   
1             AL              35957                      AL - Birmingham   
2             AL              35631                      AL - Birmingham   

    Total Discharges   Average Covered Charges   Average Total Payments   \
0                  91                 $32963.07                 $5777.24   
1                  14                 $15131.85                 $5787.57   
2                  24                 $37560.37                 $5434.95   

  Average Medicare Payments  
0                  $4763.73  
1                  $4976.71  
2                  $4453.79

In [669]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 163065 entries, 0 to 163064
Data columns (total 12 columns):
 #   Column                                Non-Null Count   Dtype 
---  ------                                --------------   ----- 
 0   DRG Definition                        163065 non-null  object
 1   Provider Id                           163065 non-null  int64 
 2   Provider Name                         163065 non-null  object
 3   Provider Street Address               163065 non-null  object
 4   Provider City                         163065 non-null  object
 5   Provider State                        163065 non-null  object
 6   Provider Zip Code                     163065 non-null  int64 
 7   Hospital Referral Region Description  163065 non-null  object
 8    Total Discharges                     163065 non-null  int64 
 9    Average Covered Charges              163065 non-null  object
 10   Average Total Payments               163065 non-null  object
 11  Average Medic

# Format names and data

In [670]:
# To remove spaces
df.columns = df.columns.str.strip()
df.rename(columns={'Average Covered Charges': 'Avg Covered Charges [$]','Average Total Payments' : 'Avg Total Payments [$]', 'Average Medicare Payments': 'Avg Medicare Payments [$]'}, inplace=True)
df.head(3)

DRG Definition  Provider Id  \
0  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10001   
1  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10005   
2  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10006   

                      Provider Name     Provider Street Address Provider City  \
0  SOUTHEAST ALABAMA MEDICAL CENTER      1108 ROSS CLARK CIRCLE        DOTHAN   
1     MARSHALL MEDICAL CENTER SOUTH  2505 U S HIGHWAY 431 NORTH          BOAZ   
2    ELIZA COFFEE MEMORIAL HOSPITAL          205 MARENGO STREET      FLORENCE   

  Provider State  Provider Zip Code Hospital Referral Region Description  \
0             AL              36301                          AL - Dothan   
1             AL              35957                      AL - Birmingham   
2             AL              35631                      AL - Birmingham   

   Total Discharges Avg Covered Charges [$] Avg Total Payments [$]  \
0                91               $32963.07               $5777.24   
1                14               $15131.85               $5787.57   
2                24               $37560.37               $5434.95   

  Avg Medicare Payments [$]  
0                  $4763.73  
1                  $4976.71  
2                  $4453.79

In [671]:

# To remove $  symbol and convert into float
df[['Avg Covered Charges [$]', 'Avg Total Payments [$]', 'Avg Medicare Payments [$]']] =  df[['Avg Covered Charges [$]', 'Avg Total Payments [$]', 'Avg Medicare Payments [$]']].replace('\$','',regex=True).astype(float)

# Answer questions

### 1. Which are the most expensive states and the cheapest ones?
The best variable to compare is **Average Total Payments** instead of **Average Covered Charges** because the last one is only the ticket price, not the real amount paid.

In [672]:
df1 = df.groupby(df['Provider State']).median()
df1.drop(df[['Provider Id',	'Provider Zip Code', 'Total Discharges']], axis=1, inplace=True)
df1.head()

Avg Covered Charges [$]  Avg Total Payments [$]  \
Provider State                                                    
AK                              31073.2                 11190.1   
AL                              21620.3                  5876.4   
AR                              19270.6                  6151.7   
AZ                              30215.7                  7671.4   
CA                              49465.1                  9378.0   

                Avg Medicare Payments [$]  
Provider State                             
AK                                 9594.9  
AL                                 4811.2  
AR                                 5165.3  
AZ                                 6471.1  
CA                                 8380.9

In [673]:
df1 = df1.astype(int)

In [674]:
sorted_df1 = df1.sort_values(by=['Avg Total Payments [$]'])

In [680]:
sorted_df1.plot_bokeh.bar(figsize=(1200,400), title='Hospital Charges in America by States')

Figure(id='227146', ...)


🔍 According to **Average Total Payments**  the three most expensive states✳️ are:
1. **Alaska** with $11190.
2. **Hawaii** with $10148.
3. **California** with $9378.  

✳️ **Washington, D.C.** is not an state, however is cosidered an special distric because is the capital city of the United States.

🔍 According to **Average Total Payments**  the three most cheapest states✳️ are:
1. **Alabama** with $5876.
2. **West Virginia** with $6102.
3. **Arkansas** with $6151.  

In [676]:
# To answer if exist some correlation between Average Covered Charges, Average Total Payments and Average Medicare Payments
df[['Avg Covered Charges [$]', 'Avg Total Payments [$]', 'Avg Medicare Payments [$]']].corr(method ='pearson')

Avg Covered Charges [$]  Avg Total Payments [$]  \
Avg Covered Charges [$]                        1.0                     0.8   
Avg Total Payments [$]                         0.8                     1.0   
Avg Medicare Payments [$]                      0.8                     1.0   

                           Avg Medicare Payments [$]  
Avg Covered Charges [$]                          0.8  
Avg Total Payments [$]                           1.0  
Avg Medicare Payments [$]                        1.0